lib/bootloader_setup.pm

```diff
@@ -1040,37 +1040,77 @@ sub change_grub_config {
 
 =head2 add_grub_cmdline_settings
 
-    add_grub_cmdline_settings($add [, $update_grub ]);
+    add_grub_cmdline_settings($add [, $update_grub , $search]);
 
 Add C<$add> into /etc/default/grub, using sed.
 C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+C<$search> if set, bypass default grub cmdline variable.
 =cut
 sub add_grub_cmdline_settings {
+    my ($add, $update_grub, $search) = @_;
+    $search //= get_cmdline_var();
+    change_grub_config('"$', " $add\"", $search, "g", $update_grub);
+}
```

Comment:
I didn't notice, that using `$search` will require to specify `$update_grub`, maybe it'd make sense to use `%args` for these. But I leave it up to you.

Reply:
I think I need to fix this issue so that it could be used safely. Please waiting the next version.

Reply:
Thanks for taking care! And sorry to bother you so much for just simple functions.

Reply:
No, I'm just perl junior, I have to learn something. and Product usually crashes at the simple functions.